app/grandchallenge/charts/static/js/charts/render_charts.mjs

```diff
@@ -41,9 +34,28 @@ const observer = new IntersectionObserver(handleInterSection, {
   threshold: 1.0,
 });
 
-
 document.addEventListener("DOMContentLoaded", function(event) {
-    for (const element of document.getElementsByClassName("vega-lite-chart-lazy")) {
+    for (const element of document.getElementsByClassName("vega-lite-chart")) {
         observer.observe(element, );
     }
 });
+
+
+function mutationObserverCallback(mutationList, observer) {
+    mutationList.forEach((mutation) => {
+        mutation.addedNodes.forEach((addedNode) => {
+            if (addedNode.nodeType !== Node.TEXT_NODE) {
+                for (const element of addedNode.getElementsByClassName("vega-lite-chart")) {
+                    if (!element.getAttribute(handledAttribute)) {
+                        renderVegaLiteChart(element);
+                        element.setAttribute(handledAttribute, "");
+                    }
+                }
+            }
+        })
+    })
+}
+
+const mutationObserver = new MutationObserver(mutationObserverCallback);
```

Comment:
`mutationObserver` and `mutationObserverCallback` should have more specific names. What do these functions do? Maybe `mutationObserverCallback` is `renderVegaChartsInAddedNodes`? And the observer is `renderVegaChartsObserver`?

Reply:
I agree. Done